test/__init__.py

```diff
@@ -82,14 +82,16 @@ def get_sample_diff(self, name: str, labels: dict[str, str] | None = None) -> fl
         """Return the increase in the metric during the context manager protocol.
 
         If it is not found, returns ``None``. If it was not found on entry,
-        returns the value on exit.
+        raises an exception.
         """
         if labels is None:
             labels = {}
         before = self._get_value(self._before, name, labels)
         after = self._get_value(self._after, name, labels)
         if before is not None and after is not None:
             return after - before
+        elif after is not None:
+            raise ValueError(f"Metric {name}{labels} did not exist at start")
```

Comment:
You can put this here/declare it so because you're confident `before is None` at this point ?

Reply:
Exactly.